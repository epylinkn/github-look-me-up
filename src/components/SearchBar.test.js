import React from 'react'
import { shallow } from 'enzyme'
import SearchBar from './SearchBar'

let wrapper
const mockHandleSubmit = jest.fn()
const props = {
  username: 'sindresorhus',
  usernameInput: React.createRef(),
  handleSubmit: mockHandleSubmit,
}

beforeEach(() => {
  wrapper = shallow(<SearchBar {...props} />)
})

it('renders correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

it('displays the username', () => {
  expect(wrapper.find('.SearchBar-username').text()).toEqual(props.username)
})

describe('When the form is submitted', () => {
  it('calls the mock handleSubmit function', () => {
    expect(mockHandleSubmit.mock.calls.length).toBe(0)
    wrapper.find('form').simulate('submit')
    expect(mockHandleSubmit.mock.calls.length).toBe(1)
  })
})
