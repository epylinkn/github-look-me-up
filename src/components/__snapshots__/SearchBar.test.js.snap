// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    handleSubmit={[Function]}
    username="sindresorhus"
    usernameInput={
        Object {
            "current": null,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
>
        <label
                className="SearchBar-label"
                htmlFor="username"
        >
                Username: 
        </label>
        <input
                id="username"
                type="text"
        />
        <button
                className="SearchBar-submit"
                type="submit"
        >
                GO!
        </button>
        <div
                className="SearchBar-title"
        >
                Public events for
                 
                <span
                        className="SearchBar-username"
                >
                        sindresorhus
                </span>
        </div>
</form>,
      "className": "SearchBar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="SearchBar-label"
            htmlFor="username"
>
            Username: 
</label>,
          <input
            id="username"
            type="text"
/>,
          <button
            className="SearchBar-submit"
            type="submit"
>
            GO!
</button>,
          <div
            className="SearchBar-title"
>
            Public events for
             
            <span
                        className="SearchBar-username"
            >
                        sindresorhus
            </span>
</div>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Username: ",
            "className": "SearchBar-label",
            "htmlFor": "username",
          },
          "ref": null,
          "rendered": "Username: ",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "username",
            "type": "text",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "GO!",
            "className": "SearchBar-submit",
            "type": "submit",
          },
          "ref": null,
          "rendered": "GO!",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Public events for",
              " ",
              <span
                className="SearchBar-username"
>
                sindresorhus
</span>,
            ],
            "className": "SearchBar-title",
          },
          "ref": null,
          "rendered": Array [
            "Public events for",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "sindresorhus",
                "className": "SearchBar-username",
              },
              "ref": null,
              "rendered": "sindresorhus",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
>
          <label
                    className="SearchBar-label"
                    htmlFor="username"
          >
                    Username: 
          </label>
          <input
                    id="username"
                    type="text"
          />
          <button
                    className="SearchBar-submit"
                    type="submit"
          >
                    GO!
          </button>
          <div
                    className="SearchBar-title"
          >
                    Public events for
                     
                    <span
                              className="SearchBar-username"
                    >
                              sindresorhus
                    </span>
          </div>
</form>,
        "className": "SearchBar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="SearchBar-label"
              htmlFor="username"
>
              Username: 
</label>,
            <input
              id="username"
              type="text"
/>,
            <button
              className="SearchBar-submit"
              type="submit"
>
              GO!
</button>,
            <div
              className="SearchBar-title"
>
              Public events for
               
              <span
                            className="SearchBar-username"
              >
                            sindresorhus
              </span>
</div>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Username: ",
              "className": "SearchBar-label",
              "htmlFor": "username",
            },
            "ref": null,
            "rendered": "Username: ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "username",
              "type": "text",
            },
            "ref": Object {
              "current": null,
            },
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "GO!",
              "className": "SearchBar-submit",
              "type": "submit",
            },
            "ref": null,
            "rendered": "GO!",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Public events for",
                " ",
                <span
                  className="SearchBar-username"
>
                  sindresorhus
</span>,
              ],
              "className": "SearchBar-title",
            },
            "ref": null,
            "rendered": Array [
              "Public events for",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "sindresorhus",
                  "className": "SearchBar-username",
                },
                "ref": null,
                "rendered": "sindresorhus",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
